packages/components/src/pages/AnalysisFindings.vue

```diff
@@ -0,0 +1,387 @@
+              </li>
+              <li v-if="numPerformanceFindings" class="performance" @click="filter('Performance')">
+                Performance: {{ numPerformanceFindings }}
+              </li>
+              <li v-if="numStabilityFindings" class="stability" @click="filter('Stability')">
+                Stability: {{ numStabilityFindings }}
+              </li>
+              <li
+                v-if="numMaintainabilityFindings"
+                class="maintainability"
+                @click="filter('Maintainability')"
+              >
+                Maintainability: {{ numMaintainabilityFindings }}
+              </li>
+            </ul>
+          </div>
+
+          <div class="findings-list">
+            <ul class="list">
+              <li>
+                <ul class="item header">
+                  <li class="finding-name" @click="sortFindings('name')"><h3>Finding Name</h3></li>
+                  <!-- TODO
+                  <li><h3>Status</h3></li>
+                  -->
+                  <li @click="sortFindings('category')"><h3>Category</h3></li>
+                </ul>
+              </li>
+              <li v-for="[ruleTitle, impactDomain, hash] in sortedFindingsInfo" :key="hash">
+                <ul class="item" @click="openFindingInfo(hash)">
+                  <li class="finding-name">{{ ruleTitle }}</li>
+                  <!-- TODO
+                  <li>New</li>
+                  -->
+                  <li>{{ impactDomain }}</li>
+                </ul>
+              </li>
+            </ul>
+          </div>
+        </div>
+      </main>
+    </section>
+  </v-quickstart-layout>
+</template>
+
+<script>
+import VQuickstartLayout from '@/components/quickstart/QuickstartLayout.vue';
+
+export default {
+  name: 'analysis-findings',
+
+  components: {
+    VQuickstartLayout,
+  },
+
+  props: {
+    findings: {
+      default: () => [],
+      type: Array,
+    },
+  },
+
+  data() {
+    return {
+      sortBy: 'name',
+      sortAscending: true,
+      sortedFindingsInfo: this.getUniqueFindings(),
+    };
+  },
+
+  watch: {
+    findings(newVal) {
+      this.findings = newVal;
+    },
+  },
+
+  methods: {
+    sortFindings(sortBy) {
+      if (this.sortBy === sortBy) {
+        this.sortAscending = !this.sortAscending;
+      }
+
+      this.sortBy = sortBy;
+      if (this.sortBy === 'category') {
+        this.sortFindingsByCategory();
+      } else if (this.sortBy === 'name') {
+        this.sortFindingsByName();
+      }
+    },
+
+    sortFindingsByName() {
+      this.sortedFindingsInfo.sort((a, b) => {
+        if (this.sortAscending) {
+          return a[0] < b[0] ? -1 : 1;
+        }
+        return a[0] < b[0] ? 1 : -1;
+      });
+    },
+
+    sortFindingsByCategory() {
+      this.sortedFindingsInfo.sort((a, b) => {
+        if (this.sortAscending) {
+          return a[1] < b[1] ? -1 : 1;
+        }
+        return a[1] < b[1] ? 1 : -1;
+      });
+    },
```

Comment:
Consider how these methods could be altered to be `computed` properties instead of state mutations.
E.g.:
```js
computed: {
  sortedFindings() {
    const sortIndex = { name: 0, category: 1 }[this.sortBy] || 0;
    return [...this.sortedFindingsInfo].sort((a, b) => {
      if (this.sortAscending) {
        return a[sortIndex] < b[sortIndex] ? -1 : 1;
      }
      return a[sortIndex] < b[sortIndex] ? 1 : -1;
    });
  }
}
```

Reply:
I refactored the page and it makes more use of `computed` properties, but I still have some sort methods. Let me know what you think of the changes.